snuba/util.py

```diff
@@ -300,7 +300,7 @@ def conditions_expr(dataset, conditions, body, depth=0):
         lhs, op, lit = conditions
 
         if (
-            lhs in ('received', 'timestamp') and
+            lhs in ('start_ts', 'finish_ts', 'received', 'timestamp') and
```

Comment:
please let's delegate this to the dataset, you have it in this method so calling a method on the dataset class is a trivial change here.

For now you can a method in dataset that preprocess a condition (lhs, op, lit) and return a tuple (lhs, op, lit)
you can give it a default implementation on dataset
and an implementation in TimeSeriesDataset that can define the columns